src/main/kotlin/app/revanced/patches/amazon/deeplinking/DeepLinkingPatch.kt

```diff
@@ -0,0 +1,27 @@
+package app.revanced.patches.amazon.deeplinking
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+
+@Patch(
+    name = "Always allow deep-linking",
+    description = "Allows the Amazon app to open Amazon links, even if it is not the default link handler for Amazon domains.",
```

Comment:
Did I understood correctly:

```suggestion
    description = "Open Amazon links, even if the app is not set to handle Amazon links.",
```

Reply:
If so, why is the patch necessary? Couldn't you set it as the handler for Amazon links?

Reply:
Amazon is special in that regard, as they query the package manager with a VIEW Intent to check the installed handlers for the currently selected marketplace region (e.g. `https://www.amazon.de).

It appears that due to the domain verification system for deep links introduced in Android 12, this query only returns a result if the link handler for the queried domain has cryptographically proven that they possess ownership over that domain, and is enabled as a link handler.

If no results are returned from the query (either because the user specifically toggled link handling off for Amazon, or if the system couldn't resolve the domain verification status), Amazon then just goes ahead and disables the Activity (via `PackageManager#setComponentEnabledSetting`). This means that even if you launch an explicit Intent which specifies exactly this Activity/Component, it can no longer be launched. 

This behavior is implemented in `com.amazon.mShop.httpUrlDeepLink.DeepLinkingModule`:

![image](https://github.com/ReVanced/revanced-patches/assets/58902674/32336acb-a9cf-4e00-8985-3ad90c68e781)

The method which this patch patches to always return true:

![image](https://github.com/ReVanced/revanced-patches/assets/58902674/f9f57e67-f27e-488c-b41a-9ec5d096c1c5)

This is the method which disables the Activity/Component:

![image](https://github.com/ReVanced/revanced-patches/assets/58902674/afe6d481-8640-41c5-96e6-11aa99cff725)